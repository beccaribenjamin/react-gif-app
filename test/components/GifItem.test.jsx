import { render,screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe( 'Probando el <GifItem/>', () => {


    const title = 'Saitama';
    const url = 'https://on-punch.com/saitma'

    test('Debe de hacer match con el snapshot', () => {

        const {container} = render( <GifItem title={title} url={url}/> )

        expect( container ).toMatchSnapshot();

    });

    test( 'Debe mostar la img con el URL y el ALT indicado', () => {
        
        render( <GifItem title={title} url={url}/> )

        screen.debug()
        /*    
        expect( screen.getByRole('img').alt ).toBe( title )
        expect( screen.getByRole('img').alt ).toBe( title )
        */

        const { src, alt } = screen.getByRole('img');

        expect( src ).toBe( url );
        expect( alt ).toBe( alt );

    } )

} )