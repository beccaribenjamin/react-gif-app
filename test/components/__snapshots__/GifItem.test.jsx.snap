// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el <GifItem/> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://on-punch.com/saitma"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
